### ISSUE TITLE
Use default generic arguments for some of the `Generic*` types

### ISSUE THREAD

**Issue by edmorley** [Malax/libcnb.rs#70]
This is a review comment split out of https://github.com/Malax/libcnb.rs/pull/28#pullrequestreview-705465664.

If this suggestion ends up being wontfix we can always close this issue out, however I didn't want it to get lost given that the PR was already merged :-)

> * For some of these `Generic*` things, I wonder if we can just have default generic arguments.
> -- @hone

**Comment by edmorley** [comment]
@Malax Is this now redundant/wontfix after #129?

**Comment by Malax** [comment]
TBH, I'm not sure what @hone meant with his comment. It might be resolved by #129, but I'm not sure.

**Comment by Malax** [comment]
I think I now know what this was about. This was a comment from the `LayerLifecycle` work where we potentially could've set a default for `P: Platform` to `GenericPlatform` for example. I think @edmorley is right and this no longer applicable until [Rust RFC2532](https://rust-lang.github.io/rfcs/2532-associated-type-defaults.html) is finalized.

Feel free to re-open if you think this is still valid.

**Comment by hone** [comment]
yeah, not possible right now until that RFC is merged, but we weren't using associated types before.
